src/github_client.py

```diff
@@ -28,12 +29,37 @@
 
 logger = logging.getLogger(__name__)
 
+# Parameters of the function decorated with retry
+ParamT = ParamSpec("ParamT")
+# Return type of the function decorated with retry
+ReturnT = TypeVar("ReturnT")
 
-def catch_http_errors(func):
-    """Catch HTTP errors and raise custom exceptions."""
+
+def catch_http_errors(func: Callable[ParamT, ReturnT]) -> Callable[ParamT, ReturnT]:
+    """Catch HTTP errors and raise custom exceptions.
+
+    Args:
+        func: The target function to catch common errors for.
+
+    Returns:
+        The decorated function.
+    """
 
     @functools.wraps(func)
-    def wrapper(*args, **kwargs):
+    def wrapper(*args: ParamT.args, **kwargs: ParamT.kwargs) -> ReturnT:
```

Comment:
Above I saw "Any" being used for `args` and `kwargs`, I propose to pick one or the other to be consistent.

Reply:
Good point! I've diffeentiated Any with ParamT according to:

ParamT: If the function needs to be generic - the return types depend on the input types.
Any: if there is a concrete type being passed to it but we don't quite know the input, for example: Charm init functions - the args don't need to be generic but to cater for the lib I've typed it as any.

Would this be an acceptable reasoning?